### PR TITLE
INT-7963 - [FIX] Exclude Scan Patterns does not exclude correctly

### PR DESCRIPTION
**Pull request by juanmafabbri**
Jira ticket: https://sonatype.atlassian.net/browse/INT-7963
CI: https://jenkins.ci.sonatype.dev/job/integrations/job/jenkins/job/feature-snapshots/job/INT-7963_Exclude_Scan_Patterns_doesnt_work/lastBuild/

The purpose of this PR is to add a fix for the use of file exclusion patterns. The files to exclude are passed as parameters with a "!" at the beginning. 
For example:

Example 1: 
```
stage('build') {
        nexusPolicyEvaluation iqStage: 'build', iqApplication: 'app',
        iqScanPatterns: [[scanPattern: '**'], [scanPattern: '!*.zip']]
        failBuildOnNetworkError: true
}
```

Example 2:

image

### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: src/main/java/org/sonatype/nexus/ci/iq/RemoteScanner.groovy [hunk 1]
```diff
@@ -93,6 +93,13 @@ class RemoteScanner
         targets = targets.toList()
         targets.add(new File(pattern))
         targets = Collections.unmodifiableList(targets)
+      } else {
+        if (scanPatterns != null || !scanPatterns.isEmpty()) {
+          def filesExcludes = scanPatterns.findAll { it.startsWith('!') }
+              .collect { it.substring(1) }.join(',')
+          advancedProperties.setProperty("fileExcludes", filesExcludes)
```
[the remark this answers is not in the document]
Comment (juanmafabbri) [reply src/main/java/org/sonatype/nexus/ci/iq/RemoteScanner.groovy:104]: This change is duplicating the fileExcludes. The idea would be to eliminate the logic above as well and directly use `advancedProperties.getProperty("fileExcludes")`?

**Review by koraytugay — APPROVED** [review]
+1 nice !

File: src/main/java/org/sonatype/nexus/ci/iq/RemoteScanner.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (eduard-tita) [reply src/main/java/org/sonatype/nexus/ci/iq/RemoteScanner.groovy]: This change does not duplicate the `fileExcludes`. 

Users can set exclude patterns in scan targets (those starting with `!`) and your code translates them to `fileExcludes` in `advancedProperties`. But users can also set directly `fileExcludes` in `advancedProperties`, which your code overwrites, so they get lost.

My suggestion above is to merge the two.

[the remark this answers is not in the document]
Comment (juanmafabbri) [reply src/main/java/org/sonatype/nexus/ci/iq/RemoteScanner.groovy]: Commits: [a6bbc32](https://github.com/jenkinsci/nexus-platform-plugin/pull/310/commits/a6bbc32f747f628a9648fffeebc941faaf3fc090) and [524dcd4](https://github.com/jenkinsci/nexus-platform-plugin/pull/310/commits/524dcd407aab776ba8a7da578927ef6ac23e4b92)

**Review by eduard-tita — APPROVED** [review]
+1

**Pull request merged**